### PR TITLE
Htx fixes

### PR DESCRIPTION
**Pull request by vaishnavibhat**
htx fixes to use Session command to connect to remote ssh

Using session commands and api to connect and configure
htx on host and peer lpars

### CODE REVIEW

File: io/net/htx_nic_devices.py [hunk 1]
```diff
@@ -12,36 +12,23 @@
...
-try:
-    import pxssh
-except ImportError:
-    from pexpect import pxssh
 
 from avocado import Test
 from avocado.utils import distro
 from avocado.utils import process
-from avocado.utils.software_manager.manager import SoftwareManager
+from avocado.utils.software_manager import SoftwareManager
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:25]: please revert this change

File: io/net/htx_nic_devices.py [hunk 2]
```diff
@@ -84,11 +71,32 @@ def setUp(self):
...
+
         self.get_peer_distro()
 
+    def get_peer_distro(self):
+        """
+        Get the distro installed on peer lpar
+        """
+        cmd = "cat /etc/os-release"
+        output = self.session.cmd(cmd)
+        output = output.stdout.decode("utf-8")
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:89]: may use `genio` avocado library for file processing

Comment (PraveenPenguin) [reply io/net/htx_nic_devices.py:89]: ah it is happnig in peer , may be we can extend  intrafce.py to have this logic which will help any file parsing in peer

File: io/net/htx_nic_devices.py [hunk 3]
```diff
@@ -109,19 +117,17 @@ def build_htx(self):
...
+                self.cancel("Unable to install the package %s on peer machine" % pkg)
+
         if self.htx_url:
             htx = self.htx_url.split("/")[-1]
             htx_rpm = self.fetch_asset(self.htx_url)
             process.system("rpm -ivh --force %s" % htx_rpm)
-            self.run_command("wget %s -O /tmp/%s" % (self.htx_url, htx))
-            self.run_command("cd /tmp")
-            self.run_command("rpm -ivh --force %s" % htx)
+            cmd = "wget %s -O /tmp/%s ; cd /tmp ; rpm -ivh --force %s" % (self.htx_url, htx, htx)
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:129]: if we don't have wget this command will fail, need to handle this scenario

File: io/net/htx_nic_devices.py [hunk 4]
```diff
@@ -263,7 +213,9 @@ def update_host_peer_names(self):
         """
         host_name = process.system_output("hostname", ignore_status=True,
                                           shell=True, sudo=True).decode("utf-8")
-        peer_name = self.run_command("hostname")[-1]
+        cmd = "hostname"
+        output = self.session.cmd(cmd)
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:217]: maybe can drop `cmd` as a variable

File: io/net/htx_nic_devices.py [hunk 5]
```diff
@@ -762,26 +721,26 @@ def ip_restore_peer(self):
...
+                self.session.cmd("echo \'%s\' > %s" % ('TYPE=Ethernet', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('BOOTPROTO=none', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('NAME=%s' % interface, file_name))
+                self.session.cmd("echo \'%s\' >> %s" % (('DEVICE=%s' % interface), file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('ONBOOT=yes', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPADDR=%s' % ip, file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('NETMASK=%s' % self.netmask, file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6INIT=yes', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6_AUTOCONF=yes', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6_DEFROUTE=yes', file_name))
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:733]: can we move this logic in avocado utility to configure the interface

File: io/net/htx_nic_devices.py [hunk 6]
```diff
@@ -762,26 +721,26 @@ def ip_restore_peer(self):
...
-                self.run_command("echo \'%s\' > %s" % ('IPADDR=%s' % ip, file_name))
-                self.run_command("echo \'%s\' >> %s" % ('NETMASK=%s' % self.netmask, file_name))
-                self.run_command("echo \'%s\' >> %s" % ('IPV6INIT=yes', file_name))
-                self.run_command("echo \'%s\' >> %s" % ('IPV6_AUTOCONF=yes', file_name))
-                self.run_command("echo \'%s\' >> %s" % ('IPV6_DEFROUTE=yes', file_name))
+                self.session.cmd("echo \'%s\' > %s" % ('IPADDR=%s' % ip, file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('NETMASK=%s' % self.netmask, file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6INIT=yes', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6_AUTOCONF=yes', file_name))
+                self.session.cmd("echo \'%s\' >> %s" % ('IPV6_DEFROUTE=yes', file_name))
```
Comment (PraveenPenguin) [inline io/net/htx_nic_devices.py:741]: here as well

**Comment by vaishnavibhat** [comment]
avocado run --test-runner runner htx_nic_devices.py -m htx_nic_devices.yaml
JOB ID     : 15ba020e49e795979c4fe8affc68d8da7571eba1
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-06-29T09.25-15ba020/job.log
 (1/3) htx_nic_devices.py:HtxNicTest.test_start;run-cb8d: PASS (766.81 s)
 (2/3) htx_nic_devices.py:HtxNicTest.test_check;run-cb8d: PASS (121.12 s)
 (3/3) htx_nic_devices.py:HtxNicTest.test_stop;run-cb8d: PASS (6.10 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-06-29T09.25-15ba020/results.html
JOB TIME   : 902.59 s

[Uploading htx.log…]()

File: io/net/htx_nic_devices.py (hunk 2 above)
Comment (abdhaleegit) [reply io/net/htx_nic_devices.py:89]: @vaishnavibhat https://github.com/avocado-framework/avocado/blob/f9ab3aaff20d5da1312984c9cb5d95b0651164fd/avocado/utils/distro.py#L298

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

**Comment by TasmiyaNalatwad** [comment]
@abdhaleegit @vaishnavibhat @PraveenPenguin 
I have picked this patch and tested on Direct connection and on SRIOV interfaces. Please find the results below

Direct connection results
[root@ltcrain41-lp4 net]# avocado run --test-runner runner htx_nic_devices.py -m htx_nic_devices.py.data/htx_nic_devices.yaml
JOB ID     : 09dbbc43f92609de440952b93bb8e882fc39b0c1
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-07-13T04.36-09dbbc4/job.log
 (1/3) htx_nic_devices.py:HtxNicTest.test_start;run-c7c4: PASS (455.72 s)
 (2/3) htx_nic_devices.py:HtxNicTest.test_check;run-c7c4: PASS (121.54 s)
 (3/3) htx_nic_devices.py:HtxNicTest.test_stop;run-c7c4: PASS (10.06 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-07-13T04.36-09dbbc4/results.html
JOB TIME   : 58


SRIOV Interfaces
[root@ltcrain41-lp4 net]# avocado run --test-runner runner htx_nic_devices.py -m htx_nic_devices.py.data/htx_nic_devices.yaml
JOB ID     : f6855eac6828455164f4752a0f14a97f048815b9
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-07-18T08.20-f6855ea/job.log
 (1/3) htx_nic_devices.py:HtxNicTest.test_start;run-c7c4: PASS (434.38 s)
 (2/3) htx_nic_devices.py:HtxNicTest.test_check;run-c7c4: PASS (124.36 s)
 (3/3) htx_nic_devices.py:HtxNicTest.test_stop;run-c7c4: PASS (10.58 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-07-18T08.20-f6855ea/results.html
JOB TIME   : 570.60 s

**Comment by abdhaleegit** [comment]
@vaishnavibhat any more changes pending before we merge this ?

File: io/net/htx_nic_devices.py (hunk 4 above)
Comment (vaishnavibhat) [reply io/net/htx_nic_devices.py:217]: We are using multiple htx commands in various parts of the code. Hence retaining the variable for readability of the code

**Comment by abdhaleegit** [comment]
@vaishnavibhat Please make sure the htx works for all below setups
NIC dedicated
vNIC
SRIOV
HNV
Veth

**Comment by vaishnavibhat** [comment]
> @vaishnavibhat Please verify the htx works for all below setups NIC dedicated vNIC SRIOV HNV bond Veth

Hi Abdul, 
@TasmiyaNalatwad has tested this script with sriov, vnic and dedicated adapters and it works as expected.

**Comment by abdhaleegit** [comment]
@vaishnavibhat @TasmiyaNalatwad In my setup I see the host and peer are of different OS (SLES/RHEL) is there a way to handle this already ?

**Review by abdhaleegit — APPROVED** [review]
The cross distro changes can be taken in next PR
